bson/primitive/objectid_test.go

```diff
@@ -172,6 +172,38 @@ func TestCounterOverflow(t *testing.T) {
 	require.Equal(t, uint32(0), objectIDCounter)
 }
 
+func TestObjectID_MarshalJSONMap(t *testing.T) {
+	type mapOID struct {
+		Map map[ObjectID]string
+	}
+
+	oid := NewObjectID()
+	expectedJSON := []byte(fmt.Sprintf(`{"Map":{%q:"foo"}}`, oid.Hex()))
+	data := mapOID{
+		Map: map[ObjectID]string{oid: "foo"},
+	}
+
+	out, err := json.Marshal(&data)
+	require.NoError(t, err)
+	require.Equal(t, expectedJSON, out)
+}
+
+func TestObjectID_UnmarshalJSONMap(t *testing.T) {
```

Comment:
Consider incorporating this test function as a test case in `TestObjectID_UnmarshalJSON`.

Reply:
I did not add it in `TestObjectID_UnmarshalJSON` because i did not see a clean way of slotting it into the table tests.